src/main/java/com/example/demo/services/LocationService.java

```diff
@@ -1,47 +1,77 @@
 package com.example.demo.services;
 
 import com.example.demo.entities.LocationEntity;
+import com.example.demo.exceptions.LocationNotFoundException;
+import com.example.demo.model.dto.CarDto;
+import com.example.demo.model.dto.LocationDto;
 import com.example.demo.repositories.LocationEntityRepository;
+import org.modelmapper.ModelMapper;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.data.domain.PageRequest;
 import org.springframework.data.domain.Pageable;
-import org.springframework.data.domain.Sort;
-import org.springframework.http.HttpStatus;
-import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
 
+import javax.persistence.Entity;
+import javax.persistence.EntityGraph;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 
 @Service
 public class LocationService {
 
     @Autowired private LocationEntityRepository locationEntityRepository;
+    @PersistenceContext
+    EntityManager em;
+    ModelMapper modelMapper = new ModelMapper();
 
     public LocationEntity createLocation(LocationEntity location)
     {
         return locationEntityRepository.save(location);
     }
 
-    public LocationEntity findLocationById(Long Id)
+    public LocationDto findLocationById(Long id)
     {
-        return locationEntityRepository.findById(Id).orElse(null);
+        if(!locationEntityRepository.findById(id).isPresent())
+            throw new LocationNotFoundException();
+        LocationDto locationDto = modelMapper.map(locationEntityRepository.findById(id).get(),LocationDto.class);
+        locationDto.setCar(locationEntityRepository.findById(id).get().getCarsInLocation());
```

Comment:
That's three calls to findById for the same Id. Could you get it once and use the single instance instead?